src/rules_pygen/rules_generator.py

```diff
@@ -0,0 +1,514 @@
+    dependency it should have 2 wheels, one for linux one for macos
+    """
+
+    def __init__(self, name, deps, extras):
+        self.name = name.replace('-', '_').lower()
+        self._deps = deps  # subdependencies
+        self._extras = extras  # TODO(Christian): implement, don't care about it right now
+
+        self.wheels = []
+
+    @property
+    def dependencies(self):
+        return sorted([
+            dep.replace('-', '_').lower() for dep in self._deps
+        ])
+
+    def verify(self, platforms: typing.Set) -> bool:
+        """Verify that this dependency has the necessary wheels."""
+        if len(self.wheels) == 1:
+            if self.wheels[0].platform == 'purelib':
+                return True
+        existing_platforms = {w.platform for w in self.wheels}
+        return platforms == existing_platforms
+
+    def add_wheel(self, wheel: WheelInfo) -> None:
+        if self.wheels:
+            existing_platforms = {w.platform for w in self.wheels}
+            if 'purelib' in existing_platforms:
+                logger.info('Not adding any more wheels, purelib library')
+                return
+            elif wheel.platform in existing_platforms:
+                logger.info('Not adding an alternative for platform: %s', wheel.platform)
+                return
+            elif wheel.platform == 'purelib':
+                logger.info('Removing existing platform wheels, preferring a purelib variant.')
+                # weird edge case: we found a wheel that has both platform-specific and
+                # purelib variants, let's prefer the purelib wheel
+                self.wheels = []
+        self.wheels.append(wheel)
+
+    def __eq__(self, other):
+        return isinstance(other, DependencyInfo) and (self.name == other.name)
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __repr__(self):
+        return self.name
+
+
+class RequirementsToBazelLibGenerator:
+    """Generator for creating Bazel rules from a requirements-file"""
+
+    def __init__(self, requirements_path: str, wheel_dir: str, output_file: str, bzl_path: str, desired_python: str):
+        self.requirements_path = requirements_path
+        self.wheel_dir = wheel_dir
+        self.output_file = output_file
+        self.bzl_path = bzl_path
+        self.desired_python = desired_python
+        self.desired_python_full = 'python{}'.format('.'.join(self.desired_python))
+
+    def run(self) -> None:
+        """Main entrypoint into builder."""
+        self._validate()
+        logger.info('Getting wheel links via pip')
+        wheel_links = self._get_wheel_links()
+        logger.info('\nParsing dependencies from wheels\n')
+        deps = self._parse_wheel_dependencies(wheel_links)
+        logger.info('\nGenerating output file\n')
+        self._gen_output_file(deps)
+
+    def _validate(self) -> None:
+        with open(self.requirements_path, 'rb') as f:
+            if '--use-wheel' in f.readlines():
```

Comment:
If `--use-wheel` is a single line in file, then it would be read by Python as `--use-wheel\n`, so this check won't work.

Reply:
good catch!